modules/spacelift/worker-pool/remote-state.tf

```diff
@@ -36,9 +36,9 @@ module "spaces" {
   version = "1.5.0"
 
   component   = var.spacelift_spaces_component_name
-  environment = try(var.spacelift_spaces_environment_name, module.this.environment)
-  stage       = try(var.spacelift_spaces_stage_name, module.this.stage)
-  tenant      = try(var.spacelift_spaces_tenant_name, module.this.tenant)
+  environment = try(var.spacelift_spaces_environment_name, var.environment)
+  stage       = try(var.spacelift_spaces_stage_name, var.stage)
+  tenant      = try(var.spacelift_spaces_tenant_name, var.tenant)
```

Comment:
I could be missing the intention, but we should not need this or any of the following changes from `module.this.tenant/stage/environment` to `var.tenant/stage/environment`

The `spacelift/worker-pool` component has a given tenant, stage, and environment. `module.this` will take those values as default. 

However, since `spacelift/worker-pool` is deployed in `core-use2-spacelift`, we need to specify a different tenant, stage, and environment for where to find `account-map`. That's the purpose of `var.account_map_tenant_name`, `var.account_map_stage_name`, and `var.account_map_environment_name` Similarly, `spacelift/spaces` has a unique stack. We also already have variables to specify where that is: `var.spacelift_spaces_{tenant/stage/environment}_name`.

What's the purpose of these new variables that `var.account_map_{tenant/stage/environment}_name` and/or `var.spacelift_spaces_{tenant/stage/environment}_name` do not accomplish?

Reply:
If you set `enabled=false`, the component will not tear down. The `module.this` causes errors. I can get an example going, but it was done to better support `enabled=false`